recipe-server/normandy/recipes/tests/test_models.py

```diff
@@ -296,6 +298,103 @@ def test_revision_id_changes(self):
         recipe.update(action=ActionFactory())
         assert recipe.revision_id != revision_id
 
+    def test_approved_revision_property(self):
+        """Ensure approved revision properties work as expected."""
+        recipe = RecipeFactory(name='first')
+        assert recipe.name == 'first'
+
+        recipe.update(name='second')
+        assert recipe.name == 'second'
+
+        approval = ApprovalRequestFactory(revision=recipe.latest_revision)
+        approval.approve(UserFactory())
+        assert recipe.name == 'second'
+
+        recipe.update(name='third')
+        assert recipe.name == 'second'
```

Comment:
it's a bit weird but basically when you `update` it creates a new "draft" so the name doesn't change till you approve the draft.